contracts/rollover/CrossCurrencyRollover.sol

```diff
@@ -0,0 +1,437 @@
+     *         the transaction will revert.
+     *
+     * @dev All whitelisted payable currencies and collateral must be whitelisted.
+     *
+     * @param sourceLoanTerms           The terms of the original loan.
+     * @param newLoanTerms              The terms of the new loan.
+     * @param borrowerNoteId            The ID of the borrowerNote for the old loan.
+     */
+    function _validateCurrencyRollover(
+        LoanLibrary.LoanTerms memory sourceLoanTerms,
+        LoanLibrary.LoanTerms memory newLoanTerms,
+        uint256 borrowerNoteId
+    ) internal view {
+        // ------------- Caller Validation -------------
+        address borrower = IPromissoryNote(borrowerNote).ownerOf(borrowerNoteId);
+
+        if (borrower != msg.sender) revert CCR_CallerNotBorrower();
+
+        // ------------- Rollover Terms Validation -------------
+        // currency must not be the same
+        if (sourceLoanTerms.payableCurrency == newLoanTerms.payableCurrency) {
+            revert CCR_SameCurrency(sourceLoanTerms.payableCurrency, newLoanTerms.payableCurrency);
+        }
+
+        // collateral address and id must be the same
+        if (sourceLoanTerms.collateralAddress != newLoanTerms.collateralAddress || sourceLoanTerms.collateralId != newLoanTerms.collateralId) {
+            revert CCR_CollateralMismatch(
+                sourceLoanTerms.collateralAddress,
+                sourceLoanTerms.collateralId,
+                newLoanTerms.collateralAddress,
+                newLoanTerms.collateralId
+            );
+        }
+
+        // ------------- New LoanTerms Validation -------------
+        originationHelpers.validateLoanTerms(newLoanTerms);
+    }
+
+    // ========================================= HELPERS ================================================
+    /**
+     * @notice Executes the rollover of a loan from one set of terms to another.
+     *         Handles the settlement of old loan amounts and the necessary funds transfer
+     *         to repay the old loan and initialize the new loan.
+     *
+     * @param oldLoanId                  The ID of the original loan to be rolled over.
+     * @param oldLoanData                The loan data of the original loan.
+     * @param lender                     The address of the new loan lender.
+     * @param newLoanTerms               The terms of the new loan.
+     * @param swapParams                 The parameters for the currency swap.
+     */
+    function _executeRollover(
+        uint256 oldLoanId,
+        LoanLibrary.LoanData memory oldLoanData,
+        address lender,
+        LoanLibrary.LoanTerms memory newLoanTerms,
+        OriginationLibrary.SwapParameters memory swapParams
+    ) internal {
+        // get funds for new loan and swap them
+        (uint256 repayAmount, LoanLibrary.FeeSnapshot memory feeSnapshot) = _processSettlement(msg.sender, lender, swapParams, newLoanTerms, oldLoanData, oldLoanId);
```

Comment:
Can you clean this up so that it is not all one line. Hard to read and almost double the line length of all the other lines